.gitignore

```diff
@@ -1,3 +1,6 @@
 node_modules
 out
 .DS_Store
+challenges*/*
+pages/*
+.idea
```

Comment:
Because `.idea` is particular to your development environment and not this project, I think it's best to remove it here and you can perhaps add it to your global `.gitignore`.